com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ShaderUtil.java

```diff
@@ -99,17 +102,105 @@ public static String stripComments(String source)
-                   data_type == ShaderDesc.ShaderDataType.SHADER_TYPE_SAMPLER3D       ||
-                   data_type == ShaderDesc.ShaderDataType.SHADER_TYPE_TEXTURE2D       ||
-                   data_type == ShaderDesc.ShaderDataType.SHADER_TYPE_UTEXTURE2D      ||
-                   data_type == ShaderDesc.ShaderDataType.SHADER_TYPE_SAMPLER         ||
-                   data_type == ShaderDesc.ShaderDataType.SHADER_TYPE_UIMAGE2D        ||
-                   data_type == ShaderDesc.ShaderDataType.SHADER_TYPE_IMAGE2D;
+        public static String compileGLSL(String shaderSource, ShaderDesc.ShaderType shaderType, ShaderDesc.Language shaderLanguage, boolean isDebug) throws CompileExceptionError {
+
+            ByteArrayOutputStream os = new ByteArrayOutputStream();
+            PrintWriter writer = new PrintWriter(os);
+
+            // Write directives from shader.
+            String line;
+            String firstNonDirectiveLine = null;
+            int directiveLineCount = 0;
+
+            Pattern directiveLinePattern = Pattern.compile("^\\s*(#|//).*");
+            Scanner scanner = new Scanner(shaderSource);
+
+            while (scanner.hasNextLine()) {
+                line = scanner.nextLine();
+                if (line.isEmpty() || directiveLinePattern.matcher(line).find()) {
+                    writer.println(line);
+                    ++directiveLineCount;
+                } else {
+                    firstNonDirectiveLine = line;
+                    break;
+                }
+            }
+
+            if (directiveLineCount != 0) {
+                writer.println();
+            }
+
+            int version;
+            boolean gles3Standard;
+            boolean gles;
+
+            if (shaderLanguage == ShaderDesc.Language.LANGUAGE_GLSL_SM430) {
+                version       = 430;
+                gles          = false;
+                gles3Standard = true;
+            } else {
+                gles = shaderLanguage == ShaderDesc.Language.LANGUAGE_GLES_SM100 ||
+                        shaderLanguage == ShaderDesc.Language.LANGUAGE_GLES_SM300;
+
+                gles3Standard = shaderLanguage == ShaderDesc.Language.LANGUAGE_GLSL_SM140 ||
+                        shaderLanguage == ShaderDesc.Language.LANGUAGE_GLES_SM300 ||
+                        shaderLanguage == ShaderDesc.Language.LANGUAGE_GLSL_SM330;
+
+                if (shaderLanguage == ShaderDesc.Language.LANGUAGE_GLES_SM300) {
+                    version = 300;
+                } else if (shaderLanguage == ShaderDesc.Language.LANGUAGE_GLSL_SM330) {
+                    version = 330;
+                } else {
+                    version = 140;
+                }
+
+                // Write our directives.
+                if (shaderLanguage == ShaderDesc.Language.LANGUAGE_GLES_SM100) {
+                    // Normally, the ES2ToES3Converter would do this
+                    writer.println("precision mediump float;");
```

Comment:
hmm, I thought it supposes to check if precision already specified if not then set default one e.g.  `mediump`, no?

Reply:
This is legacy code, the spirv toolchain does this automatically for us

Reply:
I'm not sure, this is part of the legacy pipeline so I'm not sure we should do anything about right now (or at least in this PR).